cql3/expr/expression.hh

```diff
@@ -477,6 +491,10 @@ extern expression replace_column_def(const expression&, const column_definition*
 // For example this changes token(p1, p2) < token(1, 2) to my_column_name < token(1, 2).
 extern expression replace_token(const expression&, const column_definition*);
 
+// Reursively copies e and returns it. Calls replace_candidate() on all nodes. If it returns nullopt,
+// continue with the copying. If it returns an expression, that expression replaces the current node.
```

Comment:
It would be good to mention that it replaces only LHS in `binary_operator`

Reply:
also a typo in `Reursively`

Reply:
Or make it recurse on both sides?  Existing code will still work due to upstream constraints, and we could leverage the future generalized recursive visitor.

Reply:
It's can't recurse the RHS because it's not an expression today. Once it is, of course.

Reply:
See also my comment about this above. Whatever type the rhs is (I didn't know it wasn't an expression), you can also implement recurse() for that type, even as a a no-op. Later, when rhs becomes an expressure, the right recurse() will be magically called - and you won't need to remember to fix yet another place in the code.

Reply:
recurse is a named lambda. I could add a no-op recurse_term() and call it, and when term is removed, recurse_term() won't compile any more.

Reply:
Unfortunately, that makes column_value and column_value_tuple non-leaf expressions too (they contain terms for subscripts). We should replace them with subscript{expression, expression} and make column_value a true leaf (and replace column_value_tuple with the existing tuple_constructor).